scene/2d/animated_sprite_2d.cpp

```diff
@@ -172,18 +172,23 @@ void AnimatedSprite2D::_notification(int p_what) {
 				return;
 			}
 
-			double speed = frames->get_animation_speed(animation) * Math::abs(speed_scale);
-			if (speed == 0) {
-				return; // Do nothing.
-			}
-			int last_frame = frames->get_frame_count(animation) - 1;
-
 			double remaining = get_process_delta_time();
+			int i = 0;
 			while (remaining) {
-				if (timeout <= 0) {
-					timeout = _get_frame_duration();
+				double actual_speed_scale = frames->get_animation_speed_scale(animation) * speed_scale;
+				if (backwards) {
+					actual_speed_scale *= -1;
+				}
```

Comment:
The backward flag now respects negative speed_scale as playing_backward, so there is no need for actual_speed_scale. You should understand the changes made in the mastar and make only those changes that are truly necessary.

Reply:
It seems to me that the `backwards` parameter should be removed, since in combination with the `speed_scale` of the node and the animation, it turns out to be very confusing.

Reply:
I was considering doing that in another PR to be honest. Not just that, but adding `play_backwards`, very similar to **AnimationPlayer**. But, one thing at the time, as I've said before. I know there's this feeling of being hasty because of compatibility breaking changes, but... focus on the core. I believe this PR would benefit by being just about being able to set duration for each individual frame.

Reply:
> You should understand the changes made in the mastar

Even if we decide to leave `backwards` parameter, please take a look at all the changes in the complex. Now multiplying by `speed_scale`s and checking `backwards` happens in the processing logic. `timeout` and `_get_frame_duration` are now in their primary form (without modifiers), which allows for example changing `speed_scale` without recalculating `timeout`.

Reply:
The timing of getting/chacking timeout can be changed, but for actual_speed_scale, comparison of bool values by the playing_backward flag is sufficient so there is no need to compare double values.

Reply:
Ah, I understand. It is like you are trying to allow negative FPS for SpriteFrames. In my opinion, since SpriteFrames are only played by AnimatedSprite, the FPS in SpriteFrames should be removed instead of replacing SpeedScale.

Reply:
> the FPS in SpriteFrames should be removed instead of replaced by SpeedScale

Yes, I thought about it. We can do this, but then we need the ability to multi-edit the frame duration in two ways: 1. select several frames and set the duration for all at once, 2. multiply the duration of the selected frames by a non-negative number.

Reply:
I think it would be fine to set the default value to 1s for each frames, in the same way as AnimatedTexture, and then control it with SpeedScale on the AnimatedSprite side. Multi-editing could be added in another PR.

Reply:
Done. I removed FPS.